### ISSUE TITLE
Wrong type for Call argument of function declared in a dependency has an ambiguous type name

### ISSUE THREAD

**Issue by rooksdo** [TypeCobolTeam/TypeCobol#1786]
No error is detected when calling a function defined in a dependency with an input typed argument that has an ambiguous name because the typdef is defined in both programs.

Here is how to reproduce the bug :

**Pgm1**
```cob
       identification division.
       program-id. Pgm1.
       data division.
       working-storage section.
       01 MyType01  typedef strict public pic X.
       01 myVar     type MyType01.
       procedure division.
           call Pgm2::Foo input myVar
           .
       end program Pgm1.
```
**Pgm2**
```cob
       IDENTIFICATION DIVISION.
       PROGRAM-ID. Pgm2.
       DATA DIVISION.
       WORKING-STORAGE SECTION.
       01 MyType01  typedef strict public pic X.
       PROCEDURE DIVISION.
       INIT-LIBRARY.
           .
       declare procedure Foo public
                   input myInput type MyType01.
       end-declare.
       END PROGRAM Pgm2.
```

**Comment by fm-117** [comment]
See https://github.com/TypeCobolTeam/TypeCobol/blob/b256b5189e629412f917d8599d7e57c4d14046ef/TypeCobol/Compiler/Diagnostics/TypeCobolChecker.cs#L360
We compare the `DataType` directly instead of using the `TypeDefinition` object. To fix that :
- remove the if at this line
- there is also a bug for `TypeDefinition` of built-ins, fix the code in DataType.cs to avoid creating multiple instances of TypeDefinition objects when parsing multiple files

**Comment by fm-117** [comment]
Closed with #1814.
